app/app.go

```diff
@@ -619,6 +629,17 @@ func (app *App) RegisterTendermintService(clientCtx client.Context) {
 	tmservice.RegisterTendermintService(clientCtx, app.BaseApp.GRPCQueryRouter(), app.interfaceRegistry, nil)
 }
 
+func (app *App) setPostHanders() {
+	postHandler, err := posthandler.NewPostHandler(
+		posthandler.HandlerOptions{},
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	app.SetPostHandler(postHandler)
+}
```

Comment:
Why is this snippet necessary?

Reply:
that's a good point, I don't think it is, and we could put it this is normal typical definition of the app. tbh, I don't remember where I was going with this as the base of this PR is pretty old.

not changing for now simply so I don't dismiss the current approvals, but this should be fixed when we do #331